### DISCUSSION TITLE
WSL2 Laravel Sail - Visual Studio Code editor

### DISCUSSION THREAD

**Discussion by wit3** [spatie/laravel-ignition#23] [Q&A] [answered]
Hi, it's possibile to open with the pencil icon, a file in WSL2 with laravel sail ?

**Comment by WalterWoshid** [comment, marked as answer]
@Wit3 yes, add this to your .env file:

**EDIT:**

### WSL2 - VSCode (Remote)
```env
IGNITION_EDITOR=vscode-remote
IGNITION_LOCAL_SITES_PATH="wsl+Ubuntu-20.04/home/path/to/project"
IGNITION_REMOTE_SITES_PATH=/var/www/html
```

### WSL2 - VSCode (Local)
```env
IGNITION_EDITOR=vscode
IGNITION_LOCAL_SITES_PATH=/wsl$/Ubuntu-20.04/home/walter/Projects/PHP/my-project
```

### WSL2 - PhpStorm
```env
IGNITION_EDITOR=phpstorm
IGNITION_LOCAL_SITES_PATH=//wsl$/Ubuntu-20.04/home/walter/Projects/PHP/my-project
IGNITION_REMOTE_SITES_PATH=/var/www/html
```

**Reply by wit3** [reply]
For setting vscode i use the 
`IGNITION_EDITOR=vscode`

but if use the suggested IGNITION_LOCAL_SITES_PATH like this
`IGNITION_LOCAL_SITES_PATH="//wsl.$/Ubuntu-20.04/home/wit3/code/projects/nameofproject`
or even 
`IGNITION_LOCAL_SITES_PATH=//wsl.$/Ubuntu-20.04/home/wit3/code/projects/nameofproject`

when click on the pencil the editor alert the path doesn't exist

**Reply by WalterWoshid** [reply]
Can you try it again with `//wsl$/` instead of `//wsl.$/`. You also have to refresh the page after the change

**Reply by wit3** [reply]
Same, path doesn't exists

**Reply by WalterWoshid** [reply]
What does it say in the left bottom corner, when you hover over the pen?

**Reply by wit3** [reply]
the link is this
[vscode://file/%2F%2Fwsl%24%2FUbuntu-20.04%2Fhome%2Fwit3%2Fcode%2Fprojects%2Fx3energy%2Fadascat%2Fapp%2FModels%2FUser.php:111](vscode://file/%2F%2Fwsl%24%2FUbuntu-20.04%2Fhome%2Fwit3%2Fcode%2Fprojects%2Fx3energy%2Fadascat%2Fapp%2FModels%2FUser.php:111)

I think the problem is in the Remote Wsl workspace, vscode link not reach that

**Reply by WalterWoshid** [reply]
@Wit3 Also are you using docker or are you running the project with `artisan serve`?

**Reply by wit3** [reply]
Laravel Sail ( that use docker under the hood ) 

I open the wsl workspace in Windows terminal, go in my path and do "code ." and "sail up"

**Reply by WalterWoshid** [reply]
@Wit3 Here you go, just remove the first `/`
`IGNITION_LOCAL_SITES_PATH=/wsl$/Ubuntu-20.04/home/walter/Projects/PHP/my-project`
:)

**Reply by wit3** [reply]
just removing first / open the file in a new editor not the current with the entire workspace

**Reply by wit3** [reply]
i think the problem is in vscode initial link, maybe i need vscode+remote

**Reply by WalterWoshid** [reply]
Hmmm, when you keep the workspace open, and click on the pen again, does it open a third one or in the second one? If it opens in the second workspace again, then maybe you should not open the project with `code .`, but with `File -> Open Folder...`

**Reply by WalterWoshid** [reply]
I do not use vscode+remote and it works for me

**Reply by wit3** [reply]
I've found the holy way!

```
IGNITION_EDITOR=vscode-remote
IGNITION_LOCAL_SITES_PATH="wsl+Ubuntu-20.04/home/path/to/project"
IGNITION_REMOTE_SITES_PATH=/var/www/html
```
Thanks @WalterWoshid

**Reply by WalterWoshid** [reply]
@Wit3 Very nice that you figured it out, I edited my answer, can you mark it as answered, so others can see it? :)

**Reply by nguyendachai0** [reply]
Is it Open to file in VS code, for me it only open vscode only, not go to folder contain code or open file
